src/plugins/intel_npu/src/plugin/npuw/partitioning/partitioning.cpp

```diff
@@ -563,8 +571,8 @@ void Partitioner::identifySubgraphs() {
                           return false;
                       }
                       // Sanity check
-                      NPUW_ASSERT(p1.first->get_friendly_name() != p2.first->get_friendly_name());
-                      return p1.first->get_friendly_name() < p2.first->get_friendly_name();
+                      // NPUW_ASSERT(p1.first.get_node_shared_ptr()->get_friendly_name() != p2.first.get_node_shared_ptr()->get_friendly_name());
```

Comment:
Please keep this assert

Reply:
This assertion can't be kept, because the multiple output nodes should have the same parent node.

Reply:
In this case please do
```
// In some cases the node can have multiple outputs
if (p1.first.get_node_shared_ptr()->get_friendly_name() != p2.first.get_node_shared_ptr()->get_friendly_name()) {
    p1.second->get_friendly_name() < p2.second->get_friendly_name();
}
```

Reply:
I found the `Output` has `operator==`. Does this work better?
```
bool Output<const Node>::operator==(const Output& other) const {
    return m_node == other.m_node && m_index == other.m_index;
}
```

Reply:
```suggestion
                      NPUW_ASSERT(p1.first != p2.first);
```

Reply:
Yes, should be fine